### PR TITLE
Select only searched files

### LINKED ISSUE #2075: Only select files filtered by search
### Steps to reproduce
1. Have a folder with various files.
screen shot 2016-11-10 at 13 24 50

2. Search within the folder (e.g. search for "mp3")
screen shot 2016-11-10 at 13 24 59

3. Select all files using the "select_all_files" checkbox.
screen shot 2016-11-10 at 13 25 07

### Expected behaviour
I would expect that only the files shown after searching are selected. This would e.g. simplify downloading only certain files.

### Actual behaviour
Instead of only selecting the 3 shown files, all files within the folder (also the ones filtered out by searching) are selected.

**Nextcloud version:** 10.0.1

I searched for similar issues and found this one: https://github.com/nextcloud/server/issues/1274 However, I think this issue here is a bit different as this is not about group actions. Correct me if I'm wrong.

@icewind1991 @jancborchardt

### PR DESCRIPTION
**Pull request by skjnldsv**
Fix #2075

### CODE REVIEW

**Comment by MorrisJobke** [comment]
* [ ] clicking the "select all" checkbox still selects all files and folders (I have 2 folders and 2 txt files and searched for "txt") - also a move failed by saying it could not move the folder into which I wanted to move and it also moved the other folder that was not visible in the search result

It even says "2 folder and 2 files" beside the action menu:
bildschirmfoto 2018-06-20 um 14 55 34


Yes I cleared my cache 😉

**Comment by skjnldsv** [comment]
@MorrisJobke Fixed :)

File: apps/files/js/filelist.js [hunk 1]
```diff
@@ -748,15 +748,44 @@
...
 			if (checked) {
 				for (var i = 0; i < this.files.length; i++) {
+					// a search will automatically hide the unwanted rows
+					// let's only select the matches
 					var fileData = this.files[i];
-					this._selectedFiles[fileData.id] = fileData;
-					this._selectionSummary.add(fileData);
+					var fileRow = this.$fileList.find('[data-id=' + fileData.id + ']');
+					// do not select already selected ones
+					console.log(fileRow, this._selectedFiles[fileData.id]);
```
Comment (MorrisJobke) [inline apps/files/js/filelist.js]: Sure that this should be in here 🤔 😉

Comment (skjnldsv) [reply apps/files/js/filelist.js]: Damnit I searched for old debug!! :see_no_evil:

**Review by juliusknorr — APPROVED** [review]
Works as expected now :+1:

**Review by MorrisJobke — APPROVED** [review]
Tested and works 👍

**Comment by raimund-schluessler** [comment]
Awesome, thanks @skjnldsv.

**Comment by Schmuuu** [comment]
Hello,

Is this fix only working with NC14 or should it work with NC13 as well?
Actually, I was thinking that I could just replace the changed file on my NC13.0.4 server and it perfectly solved the problem with the selection. Unfortunately, it caused another issue in NC13.0.4: Now the "Download" button in the top right corner was no longer working. And I was wondering if this will also be an issue in NC14 then or if it is only occurring in NC13. 
Just asking for confirmation, in case this would be a bug in NC14 then later.

**Comment by raimund-schluessler** [comment]
Simply replacing the file will most likely not work, since there have been other changes to it since NC 13.0.4.  You could try to apply the patch from this PR manually, though. Although I would recommend to wait for NC14 or to ask for a backport.

**Comment by Schmuuu** [comment]
> You could try to apply the patch from this PR manually, though.
Yes, this is what I did and I ran into this issue with the Download button not working.

I wanted to mention that early on, just in case this issue exists with NC14 as well and would be discovered too late (when NC14 is released already).

Anyway: I will wait until NC14 then. Thank you.

**Comment by simonbuehler** [comment]
this still doesn't work with latest stable 14.0.3, all files in direcrory are used and not only search  results

**Pull request merged**